### DISCUSSION TITLE
How to build navigation menu with both target and on click event?

### DISCUSSION THREAD

**Discussion by Dekubaka** [anvilistas/anvil-extras#315] [Q&A] [answered]
I was going though the docs for the navigation module and got a menu created and working with no issues. 

On page click events I usually have a function to close the nav menu if running from a mobile device. I tried to make a function and do:

```
menu = [
  {"text": "Home", "target": "home", "icon": "fa:home", "on_click": testing },
  {"text": "About", "target": "about", "icon": "fa:info"},
  {"text": "Contact", "target": "contact", "icon": "fa:envelope"},
]
```
But when I run it, it does not go to the target page. It only runs the "on_click" function.

What is the proper syntax or example to have both target and onclick event working?

**Comment by Dekubaka** [comment]
I was able to get around by having putting the mobile check on the target page itself when it loads, but it would great to know how to use on_click along with a target.

**Comment by meatballs** [comment, marked as answer]
The default behaviour when a link is clicked is defined by the [_default_link_click](https://github.com/anvilistas/anvil-extras/blob/a7d8d2ca8b434b23e2900c54c5aaa2d7584614e7/client_code/navigation.py#L63) function.

If you provide a different function via the `on_click` argument, it's up to you do define its behaviour. If you want it to open a form via the `target` argument, you need to include code to do that.
